File: models/customer.go
package models

import "github.com/go-ozzo/ozzo-validation"

type Customer struct {
	Id int
	Fio string `json:"fio"`
	FirstName string `json:"firstName"`
	LastName string `json:"lastName"`
	Patronymic string `json:"patronymic"`
	Gender string `json:"gender"`
	Phone string `json:"phone"`
	Email string `json:"email"`
	Address DeliveryAddress `json:"address" db:"-"`
	FiasAddress *Address `json:"fiasAddress" db:"-"`
}

func (m Customer) Validate() error {
	return validation.ValidateStruct(&m,
		validation.Field(&m.Fio, validation.Required),
		// validation.Field(&m.FirstName, validation.Required),
		/*validation.Field(&m.LastName, validation.Required),*/
		validation.Field(&m.Phone, validation.Required),
		validation.Field(&m.Email, validation.Required),
	)
}
/*
type CustomerIn struct {
	Customer
}*/

File: ui/src/app/cart/models/customer.ts
import {DaDataAddress} from "../../../../projects/kolkov/ngx-dadata/src/lib/models/data";

export interface ICustomerAddress {
  full?: string;
  city?: string;
  cityFiasId?: string;
  street?: string;
  streetFiasId?: string;
  streetType?: string;
  streetTypeFull?: string;
  house?: string;
  houseFiasId?: string;
  houseType?: string;
  houseTypeFull?: string;
  block?: string;
  blockType?: string;
  room?: string;
  entrance?: string
  floor?: string;
  code?: string;
  postalCode?: string;
}

export interface ICustomer {
  fio: string;
  firstName?: string;
  lastName?: string;
  patronymic?: string;
  gender: string;
  phone: string;
  email: string;
  address: ICustomerAddress;
  fiasAddress: DaDataAddress;
}

/*export class CustomerAddress implements ICustomerAddress {
  constructor(public city: string) {
  }
}

export class Customer implements ICustomer{

  constructor(
    public firstName: string,
    public lastName: string,
    public phone: string,
    public email: string,
    public address: CustomerAddress = new CustomerAddress("")
  ) {
  }

}*/
